src/main/scala/com/softwaremill/realworld/users/api/UserRegisterData.scala

```diff
@@ -1,11 +1,13 @@
 package com.softwaremill.realworld.users.api
 
+import sttp.tapir.Schema.annotations.validate
+import sttp.tapir.Validator
 import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder}
 
 case class UserRegisterData(
-    email: String,
-    username: String,
-    password: String
+    @validate(Validator.nonEmptyString) email: String,
+    @validate(Validator.nonEmptyString) username: String,
```

Comment:
**suggestion:** Could you also add a requirement on username length? For example 3 characters, which happens quite often in online services. It will also be a nice showcase for how to use multiple validations, and something other than `nonEmptyString`.